code/game/objects/items/devices/radio/radio.dm

```diff
@@ -599,12 +599,20 @@ var/global/list/default_medbay_channels = list(
 	else return
 
 /obj/item/device/radio/emp_act(severity)
+	frequency = get_random_freq()
+	if(listening == 1)
+		visible_message("<span class='warning'>\The [src] screeches violently!</span>")
+			
 	broadcasting = 0
 	listening = 0
 	for(var/ch_name in channels)
 		channels[ch_name] = 0
 	..()
 
+/obj/item/device/radio/proc/get_random_freq()
+	var/rand_freq = round(rand(1440, 1480), 10) + pick(1, 3, 5, 9)

```

Comment:
Use the defines for these, please. You can see them [here](https://github.com/ParadiseSS13/Paradise/blob/master/code/controllers/communications.dm#L100-L103)
`var/rand_freq = round(rand(PUBLIC_LOW_FREQ, PUBLIC_HIGH_FREQ), 10) + pick(1, 3, 5, 9)`